### PR TITLE
moving the auth earlier in the flow

### PR DESCRIPTION
**Pull request by otherview**
This should sort the docker limit rating we were having.

### CODE REVIEW

File: .ci/run_e2e_tests.sh [hunk 1]
```diff
@@ -16,7 +25,6 @@ if [[ -z ${DOCKER_USERNAME} ]]; then
     echo "Skipping Byzantine Tests because Docker Credentials were not present."
     BYZANTINE_IMAGE=""
 else
-    echo "$DOCKER_PASS" | docker login --username "$DOCKER_USERNAME" --password-stdin
```
Comment (holisticode) [inline .ci/run_e2e_tests.sh:19]: What I don't understand though is that even here there was the login *before* the pull.

So how does your PR change the situation we were having?

File: .ci/run_e2e_tests.sh [hunk 2]
```diff
@@ -1,5 +1,14 @@
 SCRIPTS_PATH=$(cd $(dirname "${BASH_SOURCE[0]}"); pwd)
 SRC_PATH=$(dirname "${SCRIPTS_PATH}")
+
+# Early auth to avoid limit rating
+# If Docker Credentials are not available skip the Byzantine Tests
```
Comment (holisticode) [inline .ci/run_e2e_tests.sh:5]: This seems to be similar to
https://github.com/ava-labs/avalanchego/pull/725/files#diff-5094e7b250ccea461e290b48addfa38d35af80538dbd9c313a91dafb2e5e0c4cR11?#728 

cc @aaronbuchwald

File: .ci/run_e2e_tests.sh (hunk 1 above)
Comment (otherview) [reply .ci/run_e2e_tests.sh:19]: Yeah I think this whole CI could do with a fresh review 🤔 

I moved the `login` to before the `bash "${SRC_PATH}"/scripts/build_image.sh` which pulls the `FROM golang:1.15.5-buster`

Comment (holisticode) [reply .ci/run_e2e_tests.sh:19]: aaaah gotcha, that's where we were pulling before authenticating. :+1:

**Review by holisticode — DISMISSED** [review]

File: .ci/run_e2e_tests.sh (hunk 2 above)
Comment (aaronbuchwald) [inline .ci/run_e2e_tests.sh:5]: Could you remove the second comment here as I think it's intended for the place below where we actually set byzantine image if the docker credentials are not as opposed to the authentication we're doing here?

Comment (otherview) [reply .ci/run_e2e_tests.sh]: Done :)

**Review by aaronbuchwald — APPROVED** [review]

**Pull request merged**